config/ocp_aws/install/fence_aws_credentials_request.yaml

```diff
@@ -0,0 +1,45 @@
+# similar to https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_00_credentials-request.yaml#L1-L6
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: openshift-aws-cloud-fencing
+  namespace: openshift-cloud-credential-operator
+spec:
+  serviceAccountNames:
+  - fence-agents-remediation-controller-manager
+  secretRef:
+    name: aws-cloud-fencing-credentials-secret
+    namespace: openshift-operators
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+# https://github.com/openshift/cloud-credential-operator/blob/master/pkg/apis/cloudcredential/v1/types_aws.go
+    statementEntries:
+    - action:
+      - ec2:StartInstances
+      - ec2:StopInstances
+      - ec2:DescribeImages
+      - ec2:DescribeInstances
+      - ec2:RunInstances
+      - ec2:TerminateInstances
+      - iam:PassRole
+      - iam:CreateServiceLinkedRole
+      effect: Allow
+      resource: "*"
+    - action:
+      - kms:Decrypt
+      - kms:Encrypt
+      - kms:GenerateDataKey
+      - kms:GenerateDataKeyWithoutPlainText
+      - kms:DescribeKey
+      effect: Allow
+      resource: '*'
+    - action:
+      - kms:RevokeGrant
+      - kms:CreateGrant
+      - kms:ListGrants
+      effect: Allow
```

Comment:
Do we need these?

Reply:
That's a very good question... 

I have begun with the credentials of [`openshift-machine-api-aws` CredentialRequest](https://github.com/openshift/machine-api-operator/blob/931f6f67c1c72597fc56104cd45a71c11f19293d/install/0000_30_machine-api-operator_00_credentials-request.yaml#L20-L61), plus  `ec2:StartInstances`, and `ec2:StopInstances`, and try to minimize the amount of actions. There might be more actions to remove, but this is where I have stopped.

I do remember that I tried only with `ec2` and `iam` actions, and the reboot failed (due to timeout of waiting for the node to turn off, probably not a permission problem...) without `kms` actions.
But with `kms` actions it was always working. Not on CI, but on Cluster Bot.

P.S. I tried using  the [power_timeout](https://manpages.ubuntu.com/manpages/focal/man8/fence_aws.8.html) flag to extend the timeout, but it didn't work as IIUC the [**fence_aws** FA](https://github.com/ClusterLabs/fence-agents/blob/main/agents/aws/fence_aws.py) doesn't use this flag (and only set it to [60 seconds by default](https://github.com/ClusterLabs/fence-agents/blob/main/agents/aws/fence_aws.py#L182)).